packages/darklang/languageTools/parser.dark

```diff
@@ -240,6 +242,83 @@ module Darklang =
+
+                let first =
+                  match (findNodeByFieldName firstChild "first") with
+                  | Some firstNode -> TypeReference.parse firstNode
+                  | None ->
+                    Stdlib.Result.Result.Error
+                      "No first node found in tuple_type_reference"
+
+                let second =
+                  match (findNodeByFieldName firstChild "second") with
+                  | Some secondNode -> TypeReference.parse secondNode
+                  | None ->
+                    Stdlib.Result.Result.Error
+                      "No second node found in tuple_type_reference"
+
+                let rest =
+                  firstChild
+                  |> findNodeByFieldName "rest"
+                  |> Stdlib.Option.map (fun restNode ->
+                    restNode.children
+                    |> Stdlib.List.chunkBySize 2L
+                    |> Builtin.unwrap
+                    |> Stdlib.List.map (fun chunk ->
+                      match chunk with
+                      | [ symbol; typeNode ] ->
+                        let typeRef = TypeReference.parse typeNode
+
+                        match typeRef with
+                        | Ok typeRef -> (symbol.sourceRange, typeRef)
+                        | Error _ ->
+                          (symbol.sourceRange,
+                           WrittenTypes.Unparseable { source = typeNode })
+
+                      | [ typeNode ] ->
+                        let typeRef = TypeReference.parse typeNode
+
+                        match typeRef with
+                        | Ok typeRef -> (Stdlib.Option.Option.None, typeRef)
+                        | Error _ ->
+                          (Stdlib.Option.Option.None,
+                           WrittenTypes.Unparseable { source = typeNode })
+                      | _ ->
+                        (Stdlib.Option.Option.None,
+                         WrittenTypes.Unparseable { source = chunk })))
+                  |> Stdlib.Option.withDefault []
+
+                let asterisk =
+                  (findNodeByFieldName firstChild "symbol_asterisk")
+                  |> Stdlib.Option.toResult
+                    "No symbol_asterisk node found in tuple_type_reference"
+
+                let rightParen =
+                  (findNodeByFieldName firstChild "symbol_right_paren")
+                  |> Stdlib.Option.toResult
+                    "No symbol_right_paren node found in tuple_type_reference"
+
+                match leftParen, first, asterisk, second, rightParen with
+                | Ok leftParen, Ok first, Ok asterisk, Ok second, Ok rightParen ->
+                  (WrittenTypes.TypeReference.Builtin.TTuple
+                    node.sourceRange
+                    first
+                    asterisk.sourceRange
+                    second
+                    rest
+                    leftParen.sourceRange
+                    rightParen.sourceRange)
```

Comment:
Do we not need the parts of TTuple to be wrapped in ()? with , separators? not sure how this is working, but we should probably add those anyway.